### PR TITLE
MPP-3866 Add guidelines to use e2e testing for stage release

### PR DESCRIPTION
**Pull request by say-yawn**
Adding e2e step on Base Load Engineer Playbook.

This PR resolves MPP-3866

How to test:

- [ ] I've added or updated relevant docs in the docs/ directory.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: docs/base-load-engineer-playbook.md [hunk 1]
```diff
@@ -56,12 +56,15 @@ another more appropriate party.
 2. SRE processes ticket to release the tag to production
 3. Update [Github Release][github-releases] to current release
 4. Monitor [Sentry Releases][sentry-releases] for new production issues
-5. (On your 1st Tuesday) [Release to stage][Release-to-stage] (tag, Github release notes)
+5. (On your 1st Tuesday) Before releasing to Stage, locally [run e2e tests][run-e2e-tests] against the latest main.
```
Comment (groovecoder) [inline docs/base-load-engineer-playbook.md:59]: suggestion (non-blocking): We don't have to run the tests locally. We can use the GitHub Action [here](https://github.com/mozilla/fx-private-relay/actions/workflows/playwright.yml) to do it.

File: docs/base-load-engineer-playbook.md [hunk 2]
```diff
@@ -56,12 +56,15 @@ another more appropriate party.
...
+   - Ensure that the e2e tests are passing.
+   - If e2e Playwright tests are flaky--fails to pass for reasons outside of legitimate Relay bug--consider making the tests more reliable by using the [locators][playwright-locators], [auto-retrying assertions][playwright-auto-retrying-assertions], or [fixtures][playwright-fixtures].
```
Comment (groovecoder) [inline docs/base-load-engineer-playbook.md:61]: suggestion (non-blocking): We could link directly to [the test engineering recap deck](https://docs.google.com/presentation/d/1dSASq9xcaA8DuQM_1_Ab6q5_ScBpvqI9NPHvovkA-wU/edit#slide=id.g276e3207c4d_1_427) which includes these techniques and many others.

**Review by groovecoder — APPROVED** [review]
Couple non-blocking suggestions.

File: docs/base-load-engineer-playbook.md (hunk 2 above)
Comment (say-yawn) [reply docs/base-load-engineer-playbook.md]: Updated on [92eb448](https://github.com/mozilla/fx-private-relay/pull/4961/commits/92eb4486073e47d7fd8c2ddb3127dc41911407d5) I referred to the doc but added a context that suggestions were improvements for FxA.

**Review by EMMLynch — APPROVED** [review]

**Review by groovecoder — APPROVED** [review]
great, thanks!

**Pull request merged**